hatchet_sdk/utils/tracing.py

```diff
@@ -0,0 +1,65 @@
+import json
+from functools import cache
+from typing import Any
+
+from opentelemetry import trace
+from opentelemetry.context import Context
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
+from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+from opentelemetry.sdk.trace import Tracer, TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+from opentelemetry.trace.propagation.tracecontext import TraceContextTextMapPropagator
+from opentelemetry.trace import NoOpTracerProvider
+
+from hatchet_sdk.loader import ClientConfig
+
+OTEL_CARRIER_KEY = "__otel_carrier"
+
+
+@cache
+def create_tracer(config: ClientConfig) -> Tracer:
+    ## TODO: Figure out how to specify protocol here
+    resource = Resource(
+        attributes={SERVICE_NAME: config.otel_service_name or "hatchet.run"}
+    )
+
+    if config.otel_exporter_oltp_endpoint and config.otel_exporter_oltp_headers:
+        processor = BatchSpanProcessor(
+            OTLPSpanExporter(
+                endpoint=config.otel_exporter_oltp_endpoint,
+                headers=config.otel_exporter_oltp_headers,
+            ),
+        )
+
+        trace_provider = TracerProvider(resource=resource)
+        trace_provider.add_span_processor(processor)
+    else:
+        trace_provider = NoOpTracerProvider()
+
+    trace.set_tracer_provider(trace_provider)
+
+    return trace.get_tracer(__name__)
+
+
+def create_carrier() -> dict[str, str]:
+    carrier = {}
+    TraceContextTextMapPropagator().inject(carrier)
+
+    return carrier
+
+
+def inject_carrier_into_metadata(
+    metadata: dict[Any, Any], carrier: dict[str, str]
+) -> dict[Any, Any]:
+    if carrier:
+        metadata[OTEL_CARRIER_KEY] = carrier
```

Comment:
this is a bit of a hack to prevent empty `__otel_carrier`s from showing up in the dashboard